python/social_bot/teacher_tasks.py

```diff
@@ -580,31 +579,23 @@ def __init__(self,
             random_range (float): the goal's random position range
             target_speed (float): the target speed runing to the ball. The agent will receive no more 
                 higher reward when its speed is higher than target_speed.
-            step_time (float): used to caculate speed of the agent
             reward_weight (float): the weight of the reward
         """
-        GoalTask.__init__(
-            self,
+        super().__init__(
             max_steps=max_steps,
             goal_name=goal_name,
             fail_distance_thresh=fail_distance_thresh,
             random_range=random_range)
```

Comment:
Pass reward_weight to super(), since it's already in GoalTask (or Task if you choose to put it there).